mantidproject/mantid
# Implement loading in refl gui preview tab

**Description of work.**

This PR implements loading of a workspace in the Preview tab on the ISIS Reflectometry interface. It also contains a lot of under the hood work for #32034:
- Loading is done via an ISIS Reflectometry preprocessing algorithm (which will be expanded with more preprocessing steps in the future) so we call this algorithm from the GUI.
- Some reorganisation has been done to support calling this preprocessing algorithm from the Preview tab; primarily the addition of a `ReflAlgorithmFactory` which has the job of converting the reduction configuration i.e. our model to a configured algorithm (i.e. the algorithm and its properties); and the `PreviewJobManager`, which is responsible for calling the `ReflAlgorithmFactory` and executing the algorithm.
- Similar changes have been partially made to the `BatchJobManager`. This has been renamed from `BatchJobRunner` because it does not directly run the algorithms; its job is more to manage their creation.
- The `BatchJobManager` behaviour could be consolidated further (i.e. move the execution of algorithms here instead of doing it via the `BatchPresenter`) but this is beyond the scope of this PR.
- `MockAlgorithm` from the test helpers has been renamed to `StubAlgorithm`, because this is not a real mock.

**To test:**

- Open the ISIS Reflectometry interface
- Set instrument to INTER
- Click the Preview tab
- Enter run number `13460` and click Load. A message should appear in the log saying `Loaded ws pointer`. This indicates the load has worked. This is work in progress so nothing else will happen at this point.
- Ensure errors such as empty run number and bad file name are handled i.e. no crashing.

Fixes #32125





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
